### PR TITLE
Feature/native dark mode support

### PR DESCRIPTION
**Pull request by NinjaSnail42**
### Notes

- To change the text color of the TextViews with attributed text, I had to do it programatically. Since this is the case, I converted the Objective-C code for those three view controllers (which I am currently working on on a different branch anyway.)
- I did not change the way that the blue color in the app appears between light and dark mode. I don't think that it looks bad in dark mode, but if we are interested in changing the color between light and dark mode, there are a couple things to know. You can do this by adding color sets, but this is not supported the on older versions of iOS that the app supports. It could also done programmatically, but will be extra work and would be best to do after the entire app is converted from Objective-C to Swift.

### Testing

- I tested on iOS Simulators with iOS 10.3 (iPhone 5s) and iOS 13 (iPhone 11 Pro Max).

### CODE REVIEW

File: SafetyPlan/GuideCopingViewController.swift [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+    
+    
+    override func viewDidLayoutSubviews() {
+        self.copingTextView.setContentOffset(CGPoint(x: 0.0, y: 0.0), animated: false)
+    }
+    
+    
+    override func didReceiveMemoryWarning() {
+        super.didReceiveMemoryWarning()
+    }
```
Comment (shayneptorres) [inline SafetyPlan/GuideCopingViewController.swift]: If you are not adding any extra/custom logic in this method and just calling `super.didReceiveMemoryWarning()` you don't need to override it. This method can be removed

File: SafetyPlan/GuideRecoveringViewContoller.swift [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+    
+    
+    override func viewDidLayoutSubviews() {
+        self.recoveringTextView.setContentOffset(CGPoint(x: 0.0, y: 0.0), animated: false)
+    }
+    
+    
+    override func didReceiveMemoryWarning() {
+        super.didReceiveMemoryWarning()
+    }
```
Comment (shayneptorres) [inline SafetyPlan/GuideRecoveringViewContoller.swift]: If you are not adding any extra/custom logic in this method and just calling `super.didReceiveMemoryWarning()` you don't need to override it. This method can be removed

File: SafetyPlan/GuideSuicidePreventionViewController.swift [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+    
+    
+    override func viewDidLayoutSubviews() {
+        self.suicidePreventionTextView.setContentOffset(CGPoint(x: 0.0, y: 0.0), animated: false)
+    }
+    
+    
+    override func didReceiveMemoryWarning() {
+        super.didReceiveMemoryWarning()
+    }
```
Comment (shayneptorres) [inline SafetyPlan/GuideSuicidePreventionViewController.swift]: If you are not adding any extra/custom logic in this method and just calling `super.didReceiveMemoryWarning()` you don't need to override it. This method can be removed

File: SafetyPlan/GuideCopingViewController.swift [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+import UIKit
+
+
+class GuideCopingViewController: UIViewController {
+    @IBOutlet weak var copingTextView: UITextView!
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        
+        self.copingTextView .scrollRangeToVisible(NSMakeRange(0, 0))
```
Comment (shayneptorres) [inline SafetyPlan/GuideCopingViewController.swift]: We can remove the extra space before the period

File: SafetyPlan/GuideRecoveringViewContoller.swift [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+import UIKit
+
+
+class GuideRecoveringViewContoller: UIViewController {
+    @IBOutlet weak var recoveringTextView: UITextView!
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        
+        self.recoveringTextView .scrollRangeToVisible(NSMakeRange(0, 0))
```
Comment (shayneptorres) [inline SafetyPlan/GuideRecoveringViewContoller.swift]: We can remove the extra space before the period

File: SafetyPlan/GuideCopingViewController.swift [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+        self.copingTextView.contentSize = self.copingTextView.sizeThatFits(self.copingTextView.frame.size)
+        
+        if #available(iOS 13.0, *) {
+            self.copingTextView.textColor = UIColor.label
+        }
+    }
+    
+    
+    override func viewDidLayoutSubviews() {
+        self.copingTextView.setContentOffset(CGPoint(x: 0.0, y: 0.0), animated: false)
```
Comment (shayneptorres) [inline SafetyPlan/GuideCopingViewController.swift:29]: Might be best to include `super.viewDidLayoutSubviews`

File: SafetyPlan/GuideRecoveringViewContoller.swift [hunk 7]
```diff
@@ -0,0 +1,36 @@
...
+        self.recoveringTextView.contentSize = self.recoveringTextView.sizeThatFits(self.recoveringTextView.frame.size)
+        
+        if #available(iOS 13.0, *) {
+            self.recoveringTextView.textColor = UIColor.label
+        }
+    }
+    
+    
+    override func viewDidLayoutSubviews() {
+        self.recoveringTextView.setContentOffset(CGPoint(x: 0.0, y: 0.0), animated: false)
```
Comment (shayneptorres) [inline SafetyPlan/GuideRecoveringViewContoller.swift:29]: Might be best to include `super.viewDidLayoutSubviews`

File: SafetyPlan/GuideSuicidePreventionViewController.swift [hunk 8]
```diff
@@ -0,0 +1,35 @@
...
+        self.suicidePreventionTextView.contentSize = self.suicidePreventionTextView.sizeThatFits(self.suicidePreventionTextView.frame.size)
+        
+        if #available(iOS 13.0, *) {
+            self.suicidePreventionTextView.textColor = UIColor.label
+        }
+    }
+    
+    
+    override func viewDidLayoutSubviews() {
+        self.suicidePreventionTextView.setContentOffset(CGPoint(x: 0.0, y: 0.0), animated: false)
```
Comment (shayneptorres) [inline SafetyPlan/GuideSuicidePreventionViewController.swift:29]: Might be best to include `super.viewDidLayoutSubviews`

File: SafetyPlan/GuideSuicidePreventionViewController.swift [hunk 9]
```diff
@@ -0,0 +1,35 @@
...
+import UIKit
+
+
+class GuideSuicidePreventionViewController: UIViewController {
+    @IBOutlet weak var suicidePreventionTextView: UITextView!
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        
+        self.suicidePreventionTextView .scrollRangeToVisible(NSMakeRange(0, 0))
```
Comment (shayneptorres) [inline SafetyPlan/GuideSuicidePreventionViewController.swift]: We can remove the extra space before the period

**Review by shayneptorres — CHANGES_REQUESTED** [review]
Just some small changes that I would recommend. These aren't breaking anything but would be good to keep app consistent

**Review by shayneptorres — APPROVED** [review]
Looks great, just make sure you fix the merge conflicts. I also agree we should add a .gitignore

**Comment by eddielement** [comment]
I just checked over it now and it looks great to me too just with a few merge conflicts to fix. Agreed on the gitignore (although I've tried and failed in the past at ignoring that- maybe there are lots of different DS_Stores?). Thanks for your input too @shayneptorres !

**Comment by NinjaSnail42** [comment]
Yeah, apologies for the merge conflict on the storyboard file. I was a dev manager on an iOS project last year for a class at University, and it is quite a pain to fix merge conflicts in the storyboard files.

**Comment by NinjaSnail42** [comment]
I fixed the merge conflicts. Had to change repositories from my fork to the actual repo. Should be good for an admin to merge now.

**Comment by eddielement** [comment]
Whew thank you for that @NinjaSnail42 !! I tried to resolve the conflicts for 30 minutes earlier, it was quite hellish, and I ended up discarding all my changes and was going to try again tonight. I made a few tweaks and I'm merging the changes in now! :)

**Pull request merged**